src/http.cc

```diff
@@ -75,10 +75,11 @@ static void copyOneHeaderFromClientsideRequestToUpstreamRequest(const HttpHeader
         HttpHeader * hdr_out, const int we_do_ranges, const Http::StateFlags &);
 
 HttpStateData::HttpStateData(FwdState *theFwdState) :
+    _peer(nullptr),
     AsyncJob("HttpStateData"),
     Client(theFwdState),
-    lastChunk(0),
     httpChunkDecoder(NULL),
+    lastChunk(0),
```

Comment:
Please undo this out of scope and incorrect (AFAICT) change: lastChunk should be initialized before httpChunkDecoder because it is declared before httpChunkDecoder.

Reply:
FYI: This complexity of initialization order is one of the reasons C++11 initialization is different, and should be used.

Reply:
Right. Moving it all to the header